nep-30.mediawiki

```diff
@@ -0,0 +1,58 @@
+<pre>
+  NEP: 30
+  Title: Contract witness verification callback
+  Author: Roman Khimov <roman@nspcc.ru>
+  Type: Standard
+  Status: Accepted
+  Created: 2024-07-25
+</pre>
+
+==Abstract==
+
+This NEP standardizes a specific method that if implemented allows to use contracts as signers for transactions and pass witness checks for their address in other code.
+
+==Motivation==
+
+Contracts may want to act as transaction senders covering for fees from GAS they have, in this case their address is the first transaction signer and there needs to be some way to verify it which is made possible by this NEP.
+
+Contract addresses may also be used in other contexts like [[nep-17.mediawiki|NEP-17]] transfers, to successfully pass witness check there contract needs to either be a method caller or be a transaction signer in which case verification of the address is needed as well.
+
+The mechanism for this verification is provided by Neo N3 since its initial version, but it was never properly standardized.
+
+==Specification==
+
+Contracts that want their addresses to be used as transaction signers MUST implement <code>verify</code> method outlined below. This method will be automatically called during transaction verification (with <code>Verification</code> trigger), its return value signifies whether verification was successful (in which case transaction is considered to be valid) or not (in which case transaction is invalid).
+
+===Methods===
+
+====verify====
+
+<pre>
+{
+  "name": "verify",
+  "safe": true,
+  "parameters": [],
+  "returntype": "bool"
+}
+</pre>
```

Comment:
Example with parameters is a must

Reply:
It's not an example, it's a part of the standard and we have a sentence about parameters below. I think have any set of parameters there would make it confusing for the reader. Maybe some implementation example would be more helpful, but the only example I can remember is Notary which is native, so not really relevant for smart contract authors.